crates/sage-cli/src/main.rs

```diff
@@ -75,7 +78,18 @@ impl Runner {
             )
         })?;
 
-        let database = parameters.database.clone().build(fasta);
+        let database = match parameters.database.prefilter {
```

Comment:
I like that this can be fully disabled. We need to make sure that if it is enabled and batch size is much larger than DB size, the results are still identical.

Reply:
Currently not the case unfortunately, but as mentioned, I believe this is mostly due to threading and we might just need to fiud the point where to deterministically re-organize the peptide list